tests/tt_metal/tt_metal/unit_tests/basic/device.cpp

```diff
@@ -277,3 +278,40 @@ TEST_F(DeviceFixture, TestDeviceToHostMemChannelAssignment) {
         EXPECT_EQ(channels.size(), device_group.size());
     }
 }
+
+// Test to ensure writing from 16B aligned L1 address to 16B aligned PCIe address works
+TEST_F(DeviceFixture, TestL1ToPCIeAt16BAlignedAddress) {
+    tt_metal::Program program = tt_metal::CreateProgram();
+    Device *device = this->devices_.at(0);
+    EXPECT_TRUE(device->is_mmio_capable());
+    CoreCoord logical_core(0, 0);
+
+    uint32_t base_l1_src_address = L1_UNRESERVED_BASE + L1_ALIGNMENT;
+    uint32_t base_pcie_dst_address = CQ_START + L1_ALIGNMENT;
+
+    uint32_t size_bytes = 2048 * 128;
+    std::vector<uint32_t> src = generate_uniform_random_vector<uint32_t>(0, UINT32_MAX, size_bytes / sizeof(uint32_t));
+    uint32_t num_16b_writes = size_bytes / L1_ALIGNMENT;
```

Comment:
implicitly relies on L1_ALIGNMENT value being 16

Reply:
will add a check for this